types/stripe/index.d.ts

```diff
@@ -5148,7 +5154,7 @@ declare namespace Stripe {
             transfer_data?: IPaymentIntentDataTransferDataOptions;
         }
 
-        type PaymentIntentCancellationReason = 'duplicate' | 'fraudulent' | 'requested_by_customer' | 'failed_invoice';
+        type PaymentIntentCancellationReason = 'duplicate' | 'fraudulent' | 'requested_by_customer' | 'abandoned';
```

Comment:
We need separate types for reasons that can be given for cancellation (which are the ones here) and what can be on the object which includes some reasons internal to Stripe.

Per the doc:
> Reason for cancellation of this PaymentIntent, either user-provided (duplicate, fraudulent, requested_by_customer, or abandoned) or generated by Stripe internally (failed_invoice, void_invoice, or automatic).

Reply:
Updated!